python/hsfs/core/training_dataset_engine.py

```diff
@@ -29,15 +30,31 @@ def __init__(self, feature_store_id):
         )
         self._tags_api = tags_api.TagsApi(feature_store_id, self.ENTITY_TYPE)
 
-    def save(self, training_dataset, feature_dataframe, user_write_options):
+    def save(self, training_dataset, features, user_write_options):
+        if isinstance(features, query.Query):
+            training_dataset._querydto = features
+            training_dataset._features = [
+                training_dataset_feature.TrainingDatasetFeature(
+                    name=label_name, label=True
+                )
+                for label_name in training_dataset.label
+            ]
```

Comment:
This part I don't understand. Don't you have only the label features in the `features` property then? But it should be the entire schema of the training dataset. Also why saving the schema in `_querydto`?

Reply:
Yes, only the labels will be sent and they will be validated in the backend.

Reply:
hm okay